types/react-jss/react-jss-tests.tsx

```diff
@@ -14,32 +15,29 @@ interface MyTheme {
   };
 }
 
-/**
- * helper function to counter typescripts type widening
- */
-function createStyles<C extends string>(styles: Styles<C>): Styles<C> {
-  return styles;
+interface ButtonProps {
+    label: string;
+    size: number;
 }
 
-const styles = (theme: MyTheme) =>
-  createStyles({
-    myButton: {
+const styles: StyleCreator<'myButton' | 'myLabel', MyTheme, ButtonProps> = (theme) =>
```

Comment:
This is not a very handy pattern. Every time I add a class key I have to change the annotation and the actual implementation. Typescript should be able to infer the types. Could you add an example where we explicitly annotate the props and the theme and see if typescript is able to infer everything correctly?

Reply:
Alternative is not very handly too, i think. Because you must set 2 types in implementation of function which receive props, everytime when i add a class key. And i must include in project helper function to counter typescripts type widening from previous version tests.
Example:

```ts
function createStyles<C extends string>(styles: Styles<C>): Styles<C> {
    return styles;
}

const styles = (theme: MyTheme) =>
    createStyles({
        myButton: (props: ButtonProps): CSSProperties => ({
            fontSize: props.size,
            color: theme.color.primary,
            margin: 1,
            "& span": {
                fontWeight: "revert"
            }
        }),
        myLabel: {
            fontStyle: "italic"
        }
    });
```
Besides, this two variants of using compatible. If we'll look at declaration, there exist ```Styles``` and ```StyleCreator```. First use with pure styles object, second with theme, without helpers. Price for this - it's problem which your described in comment.

Reply:
i agree, probably it's moку flexible. i remove my changes except for extend of type for class key

Reply:
You didn't have to necessarily remove your example. I just wanted to know if the type checker can infer everything correctly. 

Could you include some negative assertions. For example using props in the styling function that do not match the props used in the component to which they are attached to?

Reply:
you mean come back my expample and use it with ```WithSheet``` in other component for test?